api4/shared_channel.go

```diff
@@ -6,12 +6,45 @@ package api4
 import (
 	"encoding/json"
 	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
 )
 
 func (api *API) InitSharedChannels() {
+	api.BaseRoutes.SharedChannels.Handle("/{team_id:[A-Za-z0-9]+}", api.ApiSessionRequired(getSharedChannels)).Methods("GET")
 	api.BaseRoutes.SharedChannels.Handle("/remote_info/{remote_id:[A-Za-z0-9]+}", api.ApiSessionRequired(getRemoteClusterInfo)).Methods("GET")
 }
 
+func getSharedChannels(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
does it make sense to add this endpoint to local mode?

Reply:
I can't think of a reason not to.  I added a ticket for that:  https://mattermost.atlassian.net/browse/MM-33838